### PR TITLE
CE-180: use transifex api v3 for pulling translations

### LINKED ISSUE #171: CE-180: Migrate to transifex v3 api
The currently in-use version of the transifex api is getting deprecated in November 2022, so we need to migrate to the new v3 API.

### PR DESCRIPTION
**Pull request by corihudson**
### Resolves

- Resolves #171 and CE-180

### Proposed Changes

Updates https://github.com/LLK/scratch-l10n/pull/161 to the latest version of the transifex javascript library, and adds error handling and retries to the scripts added there. 

Similarly to that PR, this PR only handles pulling translation strings; pushing strings using the new API will come in a separate PR (as will changes for pulling strings for blocks).

### CODE REVIEW

File: lib/validate.js [hunk 1]
```diff
@@ -27,13 +19,14 @@ const placeholders = message => (
 );
 
 const validMessage = (message, source) => {
-    const transPlaceholders = placeholders(message);
-    const srcPlaceholders = placeholders(source);
+    const transPlaceholders = placeholders(message.toString());
+    const srcPlaceholders = placeholders(source.toString());
```
Comment (cwillisf) [inline lib/validate.js:23]: Should `.toString()` have been here the whole time? If not, what changed? Just curious :)

File: package.json [hunk 2]
```diff
@@ -15,10 +15,10 @@
         "lint:js": "eslint . --ext .js",
         "lint:json": "jshint -e .json www editor/blocks editor/extensions editor/interface editor/paint-editor",
         "lint": "npm run lint:js && npm run lint:json",
-        "pull:interface": "babel-node scripts/tx-pull-translations scratch-editor interface ./editor/interface/",
```
Comment (cwillisf) [inline package.json:18]: I'm curious about this rename: it seems like `tx-pull-editor` implies that it's only being used for `scratch-editor` but I see that it's still being used for other projects. Is the idea that these are all "editor-ish" projects in some way?

**Review by cwillisf — CHANGES_REQUESTED** [review]
Looks pretty good! I'm making it as "Request changes" but really it's more "I have questions" :)

File: package.json (hunk 2 above)
Comment (corihudson) [reply package.json:18]: From what I understand, yeah - could certainly rename it, although the rename does fit with the way the repo is structured (`pull:www` puts stuff in the www folder, `pull:editor` puts stuff in the editor folder).

Comment (cwillisf) [reply package.json:18]: Ah, that makes sense. Thanks!

**Comment by corihudson** [comment]
@chrisgarrity I took a look in other repos and it seems like only tx-push-src and build-i18n-src are used. will make sure any changes to those (if any, I don't expect things to change there) are reflected in other places as well.

**Comment by chrisgarrity** [comment]
@corihudson oh right - that makes sense. All the other repos get their translations from l10n, so only l10n needs to pull from transifex. That may be something to reconsider in the long run as the tooling for translations has gotten much better since l10n was created

**Pull request merged**